CorsixTH/Lua/research_department.lua

```diff
@@ -478,6 +478,17 @@ function ResearchDepartment:improveMachine(machine)
     research_info.start_strength = research_info.start_strength
       + improve_rate
     research_info.strength_imp = research_info.strength_imp + 1
+    if self.world.ui.app.config.new_machine_extra_info then
+      -- Update dynamic info for any machines that now have improved strength.
+      for _, room in ipairs(self.world.rooms) do
```

Comment:
Make a new function `updateMachinesDynamicInfo` and call that function here.

In that way, other causes for updating dynamic info can re-use this code. In that way, we 're avoiding to duplicate code.

Reply:
Note that `self.world.rooms` is all rooms in the world, not just in the player hospital.

You may want to verify the hospital of the room to limit the machines that you touch.

Reply:
Done and inside the room loop added "if room.hospital == self.hospital then"